### ISSUE TITLE
ERROR: LoadError: UndefVarError: `@main` not defined in `Baysor`

### ISSUE THREAD

**Issue by bednarsky** [kharchenkolab/Baysor#143]
Hi, and thanks for the important package! I am trying to install baysor on my HPC for running [sopa](https://github.com/gustaveroussy/sopa/), where I have to install baysor separately (just adding this info for the case the above commands mess something up). I follow [your instructions](https://github.com/kharchenkolab/Baysor), slightly adapting them so I can run the installation from a script without prompts. 

```bash
source ~/miniconda_link/etc/profile.d/conda.sh

cd ~/projects/project_name/sopa
mamba create --name sopa python=3.10
conda activate sopa
pip install -e ".[snakemake,baysor,tangram]" 

# need to separately install baysor (in addition to dependencies installed above)
# install julia (-s -- -y is to be non-interactive)
curl -fsSL https://install.julialang.org | sh -s -- -y

# reload path env variable so that it finds julia
source ~/.bashrc

# this will be created at ~/.julia/bin/baysor, need to make symlink if you don't like that location
julia -e 'using Pkg; Pkg.add(PackageSpec(url="https://github.com/kharchenkolab/Baysor.git")); Pkg.build()'
```

I get this error: 

>     Updating git-repo `https://github.com/kharchenkolab/Baysor.git`
>     Updating registry at `~/.julia/registries/General.toml`
>    Resolving package versions...
>   No Changes to `/path/to/user/.julia/environments/v1.11/Project.toml`
>   No Changes to `/path/to/user/.julia/environments/v1.11/Manifest.toml`
> Precompiling project...
>   ✗ Baysor
>   302 dependencies successfully precompiled in 175 seconds. 40 already precompiled.
> 
> The following 1 direct dependency failed to precompile:
> 
> Baysor
> 
> Failed to precompile Baysor [cc9f9468-1fbe-11e9-0acf-e9460511877c] to "~/.julia/compiled/v1.11/Baysor/jl_1wAons".
> WARNING: both Comonicon and Base export "@main"; uses of it in module Baysor must be qualified
> ERROR: LoadError: UndefVarError: `@main` not defined in `Baysor`
> Suggestion: check for spelling errors or missing imports.
> Hint: a global variable of this name also exists in Base.
> Stacktrace:
>  [1] top-level scope
>    @ :0
>  [2] include
>    @ ./Base.jl:557 [inlined]
>  [3] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
>    @ Base ./loading.jl:2806
>  [4] top-level scope
>    @ stdin:4
> in expression starting at ~/.julia/packages/Baysor/0jqIq/src/Baysor.jl:24
> in expression starting at ~/.julia/packages/Baysor/0jqIq/src/Baysor.jl:1
>     Building HDF5 ──→ `~/.julia/scratchspaces/44cfe95a-1eb2-52ea-b672-e2afdf69b78f/114e20044677badbc631ee6fdc80a67920561a29/build.log`
>     Building Baysor → `~/.julia/scratchspaces/44cfe95a-1eb2-52ea-b672-e2afdf69b78f/e641fb17189f28e018ca48efcd1b9b9e19c137e0/build.log`
> ERROR: Error building `Baysor`:
> Precompiling Baysor...
> Info Given Baysor was explicitly requested, output will be shown live
> [ Info: disable lazy package loading: environment variable `LazyModules_lazyload=false` is detected
> WARNING: both Comonicon and Base export "@main"; uses of it in module Baysor must be qualified
> ERROR: LoadError: UndefVarError: `@main` not defined in `Baysor`
> Suggestion: check for spelling errors or missing imports.
> Hint: a global variable of this name also exists in Base.
> Stacktrace:
>  [1] top-level scope
>    @ :0
>  [2] include
>    @ ./Base.jl:557 [inlined]
>  [3] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
>    @ Base ./loading.jl:2806
>  [4] top-level scope
>    @ stdin:4
> in expression starting at ~/.julia/packages/Baysor/0jqIq/src/Baysor.jl:24
> in expression starting at ~/.julia/packages/Baysor/0jqIq/src/Baysor.jl:1
> in expression starting at stdin:4
>            ✗ Baysor
>   0 dependencies successfully precompiled in 20 seconds. 342 already precompiled.
> 
> ERROR: LoadError: The following 1 direct dependency failed to precompile:
> 
> Baysor
> 
> Failed to precompile Baysor [cc9f9468-1fbe-11e9-0acf-e9460511877c] to "~/.julia/compiled/v1.11/Baysor/jl_yNDAss".
> [ Info: disable lazy package loading: environment variable `LazyModules_lazyload=false` is detected
> WARNING: both Comonicon and Base export "@main"; uses of it in module Baysor must be qualified
> ERROR: LoadError: UndefVarError: `@main` not defined in `Baysor`
> Suggestion: check for spelling errors or missing imports.
> Hint: a global variable of this name also exists in Base.
> Stacktrace:
>  [1] top-level scope
>    @ :0
>  [2] include
>    @ ./Base.jl:557 [inlined]
>  [3] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
>    @ Base ./loading.jl:2806
>  [4] top-level scope
>    @ stdin:4
> in expression starting at ~/.julia/packages/Baysor/0jqIq/src/Baysor.jl:24
> in expression starting at ~/.julia/packages/Baysor/0jqIq/src/Baysor.jl:1
> in expression starting at stdin:
> in expression starting at ~/.julia/packages/Baysor/0jqIq/deps/build.jl:3
> Stacktrace:
>   [1] pkgerror(msg::String)
>     @ Pkg.Types /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Types.jl:68
>   [2] (::Pkg.Operations.var"#67#74"{Bool, Pkg.Types.Context, String, Pkg.Types.PackageSpec, String})()
>     @ Pkg.Operations /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1177
>   [3] withenv(::Pkg.Operations.var"#67#74"{Bool, Pkg.Types.Context, String, Pkg.Types.PackageSpec, String}, ::Pair{String, String}, ::Vararg{Pair{String}})
>     @ Base ./env.jl:265
>   [4] (::Pkg.Operations.var"#118#123"{String, Pkg.Types.EnvCache, Bool, Bool, Bool, Pkg.Operations.var"#67#74"{Bool, Pkg.Types.Context, String, Pkg.Types.PackageSpec, String}, Pkg.Types.PackageSpec})()
>     @ Pkg.Operations /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1911
>   [5] with_temp_env(fn::Pkg.Operations.var"#118#123"{String, Pkg.Types.EnvCache, Bool, Bool, Bool, Pkg.Operations.var"#67#74"{Bool, Pkg.Types.Context, String, Pkg.Types.PackageSpec, String}, Pkg.Types.PackageSpec}, temp_env::String)
>     @ Pkg.Operations /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1769
>   [6] (::Pkg.Operations.var"#116#121"{Dict{String, Any}, Bool, Bool, Bool, Pkg.Operations.var"#67#74"{Bool, Pkg.Types.Context, String, Pkg.Types.PackageSpec, String}, Pkg.Types.Context, Pkg.Types.PackageSpec, String, String})(tmp::String)
>     @ Pkg.Operations /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1878
>   [7] mktempdir(fn::Pkg.Operations.var"#116#121"{Dict{String, Any}, Bool, Bool, Bool, Pkg.Operations.var"#67#74"{Bool, Pkg.Types.Context, String, Pkg.Types.PackageSpec, String}, Pkg.Types.Context, Pkg.Types.PackageSpec, String, String}, parent::String; prefix::String)
>     @ Base.Filesystem ./file.jl:819
>   [8] mktempdir(fn::Function, parent::String)
>     @ Base.Filesystem ./file.jl:815
>   [9] mktempdir
>     @ ./file.jl:815 [inlined]
>  [10] #sandbox#115
>     @ /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1825 [inlined]
>  [11] build_versions(ctx::Pkg.Types.Context, uuids::Set{Base.UUID}; verbose::Bool)
>     @ Pkg.Operations /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1158
>  [12] build_versions
>     @ /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1076 [inlined]
>  [13] build(ctx::Pkg.Types.Context, uuids::Set{Base.UUID}, verbose::Bool)
>     @ Pkg.Operations /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/Operations.jl:1016
>  [14] build(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; verbose::Bool, kwargs::@Kwargs{io::IOContext{IO}})
>     @ Pkg.API /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/API.jl:1104
>  [15] build(pkgs::Vector{Pkg.Types.PackageSpec}; io::IOContext{IO}, kwargs::@Kwargs{})
>     @ Pkg.API /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/API.jl:159
>  [16] build(pkgs::Vector{Pkg.Types.PackageSpec})
>     @ Pkg.API /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/API.jl:148
>  [17] build(; name::Nothing, uuid::Nothing, version::Nothing, url::Nothing, rev::Nothing, path::Nothing, mode::Pkg.Types.PackageMode, subdir::Nothing, kwargs::@Kwargs{})
>     @ Pkg.API /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/API.jl:174
>  [18] build()
>     @ Pkg.API /path/to/user/.julia/juliaup/julia-1.11.0+0.x64.linux.gnu/share/julia/stdlib/v1.11/Pkg/src/API.jl:165
>  [19] top-level scope
>     @ none:1

Would be really grateful for help!

**Comment by bednarsky** [comment]
Reinstalling under `Julia/1.10.0-linux-x86_64` seems to have fixed the issue

**Comment by VPetukhov** [comment]
@bednarsky , I didn't test it on Julia 1.11 yet (it was released just a 1-2 weeks ago). But in the changelog they wrote something about changes to this macro. So I suppose there are some breaking changes there. Will fix it in the next release.

**Comment by VPetukhov** [comment]
The changes are now released in v0.7.1. While this problem is fixed, there are other compatibility issues with julia v1.11, which I can't fix. So now Baysor is restricted to v1.10. See [more info here](https://kharchenkolab.github.io/Baysor/dev/installation/#Install-as-a-Julia-package).
